objects-revisited/main.js

```diff
@@ -0,0 +1,123 @@
+const business = {
+  opens: '7am',
+  closes: '8pm',
+  totalEmployees: 4,
+  daysOpen: ['M', 'Tu', 'W', 'Th', 'F'],
+  employees: {
+    marshall: {
+      position: 'c.e.o',
+      daysOfWeekWorking: ['Tu', 'W', 'Th'],
+    },
+    nicholas: {
+      position: 'marketing director',
+      daysOfWeekWorking: ['M', 'W', 'F'],
+    },
+    justin: {
+      position: 'administrative assistant',
+      daysOfWeekWorking: ['M', 'Tu', 'W'],
+    },
+    frankie: {
+      position: 'project manager',
+      daysOfWeekWorking: ['W', 'Th', 'F'],
+    },
+  },
+};
+
+const { daysOpen, employees } = business;
+const employeePositions = [
+  'salesperson',
+  'software engineer',
+  ' c.f.o.',
+  'c.o.o.',
+];
+
+// add weekends to employee schedules
+function addWeekends(object) {
+  const weekends = ['Sa', 'Su'];
+  business.daysOpen = [...daysOpen, ...weekends];
+  for (const employee in object) {
+    object[employee].daysOfWeekWorking =
+      object[employee].daysOfWeekWorking.concat(weekends);
+  }
+}
+
+const createEmployeeObj = {};
```

Comment:
I would think you could do this without having this global `createEmployeeObj` object. Instead, declare it locally in the functions below (`addEmployees`) and pass it as a parameter to your various functions.

Doing it this way would make the code easier to extend and maintain. Global objects often lead to bugs and confusion.